src/ui/actions/timeline.ts

```diff
@@ -256,34 +237,44 @@ export function seek(
     } else {
       ThreadFront.timeWarp(point, time, hasFrames);
     }
+    dispatch(setTimelineToTime(time));
     return true;
   };
 }
 
 export function seekToTime(targetTime: number): UIThunkAction {
-  return dispatch => {
+  return async (dispatch, _getState, { ThreadFront }) => {
     if (targetTime == null) {
       return;
     }
 
-    const event = mostRecentPaintOrMouseEvent(targetTime);
+    const nearestEvent = mostRecentPaintOrMouseEvent(targetTime) || { point: "", time: Infinity };
+    const pointNearTime =
+      // @ts-ignore
+      (await sendMessage("Session.getPointNearTime", { time: targetTime }, ThreadFront.sessionId!)) // @ts-ignore
+        .point;
 
-    if (event) {
-      // Seek to the exact time provided, even if it does not match up with a
-      // paint event. This can cause some slight UI weirdness: resumes done in
-      // the debugger will be relative to the point instead of the time,
-      // so e.g. running forward could land at a point before the time itself.
-      // This could be fixed but doesn't seem worth worrying about for now.
-      dispatch(seek(event.point, targetTime, false));
+    if (Math.abs(pointNearTime.time - targetTime) > Math.abs(nearestEvent.time - targetTime)) {
+      dispatch(seek(nearestEvent.point, targetTime, false));
+    } else {
+      // I would prefer that we also use pointNearTime.time here, for accuracy,
+      // but it would be super annoying when it is off. Maybe when we have a
+      // more exact method.
+      dispatch(seek(pointNearTime.point, targetTime, false));
```

Comment:
I think we should try using `pointNearTime.time` here, otherwise we're not fixing #4289, we're just reducing the size of the lie in the Timeline. That lie has caused so much confusion that I consider it worse than showing the truth (i.e. the time that you're *really* paused at) to the user, even if the truth can also be confusing.
Likewise we should try using `nearestEvent.time` above.

Reply:
We talked about this, since it's still possible for the point to be noticeably different than the time, we're going to continue to lie, for now, but I will make a backend issue for making `getPointNearTime` more accurate, and hopefully after that we can stop lying 🙃